web3/providers/async_ipc.py

```diff
@@ -0,0 +1,166 @@
+import asyncio
+from concurrent.futures import (
+    ThreadPoolExecutor,
+)
+import errno
+from json import (
+    JSONDecodeError,
+)
+import logging
+from pathlib import (
+    Path,
+)
+import sys
+import threading
+from types import (
+    TracebackType,
+)
+from typing import (
+    Any,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+)
+
+from web3._utils.async_caching import (
+    async_lock,
+)
+from web3.types import (
+    RPCEndpoint,
+    RPCResponse,
+)
+
+from .async_base import (
+    AsyncJSONBaseProvider,
+)
+from .ipc import (
+    get_default_ipc_path,
+    has_valid_json_rpc_ending,
+)
+
+
+async def async_get_ipc_socket(
+    ipc_path: str, timeout: float = 2.0
+) -> Tuple[asyncio.StreamReader, asyncio.StreamWriter]:
+    if sys.platform == "win32":
+        # On Windows named pipe is used. Simulate socket with it.
+        from web3._utils.windows import (
+            NamedPipe,
+        )
+
+        return NamedPipe(ipc_path)
+    else:
+        return await asyncio.open_unix_connection(ipc_path)
+
+
+class PersistantSocket:
```

Comment:
persistant -> persistent

```suggestion
class PersistentSocket:
```

I think this is copied from the `PersistantSocket` defined in the synchronous `IPCProvider`. Perhaps `AsyncPersistentSocket` (or `PersistentAsyncSocket`?) might be best in case it's ever imported anywhere then we don't have the same class name imported from two separate modules? Maybe we can fix the spelling on that one in `v7` too 😅 .

Reply:
Ha thanks. Yes it is 😆